pkg/agent/manager/manager.go

```diff
@@ -0,0 +1,219 @@
+package manager
+
+import (
+	"crypto/ecdsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/spiffe/spire/pkg/agent/manager/cache"
+	"github.com/spiffe/spire/pkg/common/selector"
+	"github.com/spiffe/spire/proto/common"
+
+	tomb "gopkg.in/tomb.v2"
+)
+
+// Cache Manager errors
+var (
+	ErrNotCached         = errors.New("bundle not cached")
+	ErrPartialResponse   = errors.New("partial response received")
+	ErrUnableToGetStream = errors.New("unable to get a stream")
+)
+
+// Manager provides cache management functionalities for agents.
+type Manager interface {
+	// Start starts the manager. It blocks until fully initialized.
+	Start() error
+
+	// Shutdown blocks until the manager stops.
+	Shutdown()
+
+	// Subscribe returns a channel on which cache entry updates are sent
+	// for a particular set of selectors.
+	Subscribe(key cache.Selectors, done chan struct{}) chan *cache.WorkloadUpdate
+
+	// MatchingEntries takes a slice of selectors, and iterates over all the in force entries
+	// in order to find matching cache entries. A cache entry is matched when its RegistrationEntry's
+	// selectors are included in the set of selectors passed as parameter.
+	MatchingEntries(selectors []*common.Selector) []cache.Entry
+
+	// Stopped returns a channel on which the receiver can block until
+	// the manager stops running.
+	Stopped() chan struct{}
+
+	// Err returns the reason why the manager stopped running. If this returns
+	// nil, then the manager was stopped externally by calling its Shutdown() method.
+	Err() error
+}
+
+type manager struct {
+	c     *Config
+	t     *tomb.Tomb
+	cache cache.Cache
+
+	stopped chan struct{}
```

Comment:
feels a bit weird to have a stopped channel AND use tomb... can't we use t.Alive() or t.Dead() instead?

Reply:
Absolutely right. Will change this.